packages/dialtone-vue2/components/empty_state/empty_state.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div :class="emptyStateClasses">
+    <template v-if="showIllustration">
+      <dt-icon
+        v-if="showIcon"
+        :name="iconName"
+        size="800"
+      />
+
+      <dt-illustration
+        v-if="showIllustrationComponent"
+        :name="illustrationName"
+      />
+    </template>
+
+    <h1 v-if="headerText">
+      {{ headerText }}
+    </h1>
+
+    <p v-if="bodyText">
+      {{ bodyText }}
+    </p>
+
+    <slot name="body" />
+  </div>
+</template>
+
+<script>
+import { EMPTY_STATE_SIZE_MODIFIERS } from './empty_state_constants.js';
+import { DtIllustration, ILLUSTRATION_NAMES } from '@/components/illustration';
+import { DtIcon, ICON_NAMES } from '@/components/icon';
+
+export default {
+  name: 'DtEmptyState',
+
+  components: {
+    DtIllustration,
+    DtIcon,
+  },
+
+  props: {
+    /**
+     * The empty state size.
+     * @values 'sm', 'md', 'lg'
+     */
+    size: {
+      type: String,
+      default: 'lg',
+      validator: (s) => Object.keys(EMPTY_STATE_SIZE_MODIFIERS).includes(s),
+    },
+
+    /**
+     * The illustration name in kebab-case
+     * @type {String}
+     */
+    illustrationName: {
+      type: String,
+      default: null,
+      validator: (name) => ILLUSTRATION_NAMES.includes(name),
+    },
+
+    /**
+     * The icon name in kebab-case
+     * @type {String}
+     */
+    iconName: {
+      type: String,
+      default: null,
+      validator: (name) => ICON_NAMES.includes(name),
+    },
```

Comment:
So in the ticket we didn't have separate props for icon and illustration, they were both supposed to be combined into one. Is there a reason we weren't able to do it this way?

Perhaps calling it illustration is confusing if it's combined like this. Maybe we just call it "imageName". Doing it this way would also fix the confusion about "showIllustration". right now it's confusing because that prop only hides the illustration and not the icon.

Reply:
Ye I bring it to discussion before starting empty state regarding have a separated illustration component to handle it itself. It is more easy and clear to have it separated, less complicated than mapping if it is a illustration name or icon name and avoid issues if they have the same name.
Then we agree on have as it is now.

`showIllustration` hides both icon and illustration component, not just illustration.
I was thinking also to have it as `image` or some other synonym, but I didnt like the idea of having another name running around `image` `illustration` `icons`. 

I dont see a big deal here or a confusion that can bring wrong usability of the component.
Since you cannot have illustration or icons showing at the same time. And it will be "strange" to have some dynamic usage that you want to use this prop.

We have a comment previously regarding the usage of this props and it seems like we can get rid of it.
At this point it is there just in case....

Reply:
> It is more easy and clear to have it separated, less complicated than mapping if it is a illustration name or icon name and avoid issues if they have the same name.

It's easier for us but not easier for our users. 

Anyways I'm okay with it being like this as long as it's clearly documented. Lets just add to the comment on illustrationName "This only displays when size is medium or large"

and on iconName "This only displays when size is small" so the user knows immediately when looking at that prop.